bin/porta

```diff
@@ -451,29 +451,49 @@ module Porta
     protected
 
     def podman?
-      system('podman --version > /dev/null 2>&1')
+      return @podman if defined? @podman
+
+      @podman = system('podman --version > /dev/null 2>&1')
+    end
+
+    def need_to_add_internal_hosts?
+      return @need_to_add_internal_hosts if defined? @need_to_add_internal_hosts
+
+      @need_to_add_internal_hosts = !system("#{container_tool} run --rm --entrypoint /bin/bash #{options[:porxy_image]} -c 'getent ahostsv4 #{container_default_internal_host} > /dev/null'")
     end
 
     def container_tool
       @container_tool ||= podman? ? 'podman' : 'docker'
     end
 
+     def container_default_internal_host
+      @container_internal_host ||= podman? ? "host.containers.internal" : "host.docker.internal"
+    end
+
     def container_internal_host
-      options[:container_internal_host] ||= container_local_host_ip
+      options[:container_internal_host] ||= need_to_add_internal_hosts? ? container_default_gw : container_default_internal_host
     end
 
-    def container_local_host_ip(image: nil)
-      gw_ip_cmd = %{printf "%d.%d.%d.%d" $(awk '$2 == 00000000 && $7 == 00000000 { for (i = 8; i >= 2; i=i-2) { print "0x" substr($3, i-1, 2) } }' /proc/net/route)}
-      `#{container_tool} run --rm --entrypoint /bin/bash #{image || options[:porxy_image]} -c #{gw_ip_cmd.shellescape}`
+    # with docker and podman/slirp4netns the returned gw is the host OS
+    # on podman/pasta though, this is the local network gw so not really useful
+    def container_default_gw(image: nil)
+      gw_ip_cmd = %{printf "%d.%d.%d.%d" $(awk '$2 == 00000000 && $8 == 00000000 { for (i = 8; i >= 2; i=i-2) { print "0x" substr($3, i-1, 2) } }' /proc/net/route)}
+      `#{container_tool} run #{slirp4netns_arg} --rm --entrypoint /bin/bash #{image || options[:porxy_image]} -c #{gw_ip_cmd.shellescape}`
     end
 
+    def slirp4netns_arg
+      return @slirp4netns if defined @slirp4netns
```

Comment:
```suggestion
      return @slirp4netns if defined? @slirp4netns
```
It crashes

Reply:
Good catch! But how comes you reach this code path on Fedora 40? Did you run with docker or podman 4 here?

Reply:
I honestly don't know, I have Podman 5 on Fedora 40, and when I ran `porta deps` it immediately got crashed.

Reply:
I would be happy to have a call and debug it, to understand why it works like this. According to my investigation, podman 5 shouldn't reach this code unless you have applied some custom configuration. While using `slirp4netns` is supposed to work on podman 5, it is not the default and shouldn't be necessary.

At least I think you verified that on podman 4 the script should still work.

Reply:
We can do the call tomorrow